### PR TITLE
Kafka with Strimzi (on Kubernetes)

### PR DESCRIPTION
**Pull request by xsanm**
1. Creating and Deploying a Strimzi Kafka Cluster
2. Creating Producer and Consumer apps using Go and Scala and deploying on Kubernetes
3. Monitoring our Strimzi Kafka Cluster with Prometheus and Grafana

### CODE REVIEW

**Comment by DawidGrapa** [comment]
Missing bracket in PR name ...

**Comment by xsanm** [comment]
> Missing bracket in PR name ...

fixed :)

File: resources/kafka-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+    topicOperator: {}
+    userOperator: {}
+  kafkaExporter:
+    topicRegex: ".*"
+    groupRegex: ".*"
+    logging: debug
+    enableSaramaLogging: true
+    readinessProbe:
+      initialDelaySeconds: 15
+      timeoutSeconds: 5
```
Comment (DawidGrapa) [inline resources/kafka-metrics.yaml:72]: why not 4.99 s? performance would be better

File: resources/kafka-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+      name: kafka_server_$1_connections_tls_info
+      type: GAUGE
+      labels:
+        cipher: "$2"
+        protocol: "$3"
+        listener: "$4"
+        networkProcessor: "$5"
+    - pattern: kafka.server<type=(.+), clientSoftwareName=(.+), clientSoftwareVersion=(.+), listener=(.+), networkProcessor=(.+)><>connections
+      name: kafka_server_$1_connections_software
+      type: GAUGE
```
Comment (DawidGrapa) [inline resources/kafka-metrics.yaml:112]: GUAG

File: resources/kafka-persistent.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      transaction.state.log.min.isr: 2
+      default.replication.factor: 3
+      min.insync.replicas: 2
+      inter.broker.protocol.version: "3.4"
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
```
Comment (DawidGrapa) [inline resources/kafka-persistent.yaml:30]: man I do not have such space

File: resources/prometheus-rules.yaml [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    role: alert-rules
+    app: strimzi
+  name: prometheus-k8s-rules
+spec:
+  groups:
+  - name: kafka
+    rules:
+    - alert: KafkaRunningOutOfSpace
+      expr: kubelet_volume_stats_available_bytes{persistentvolumeclaim=~"data(-[0-9]+)?-(.+)-kafka-[0-9]+"} * 100 / kubelet_volume_stats_capacity_bytes{persistentvolumeclaim=~"data(-[0-9]+)?-(.+)-kafka-[0-9]+"} < 15
+      for: 10s
```
Comment (DawidGrapa) [inline resources/prometheus-rules.yaml:14]: why 10?

**Review by DawidGrapa — APPROVED** [review]

**Comment by adrianryt** [comment]
🤯

**Comment by xsanm** [comment]
@Atheam what do you think, Doug? 😳

**Comment by Atheam** [comment]
LGTM 👍

**Comment by wawel37** [comment]
Dobrze chłopaki robią, pozdro legnica

**Comment by wawel37** [comment]
Zydzi

File: resources/kafka-metrics.yaml [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+       listener: "$2"
+       networkProcessor: "$3"
+    - pattern: kafka.server<type=(.+), listener=(.+), networkProcessor=(.+)><>(.+)
+      name: kafka_server_$1_$4
+      type: GAUGE
+      labels:
+       listener: "$2"
+       networkProcessor: "$3"
+    # Some percent metrics use MeanRate attribute
+    # Ex) kafka.server<type=(KafkaRequestHandlerPool), name=(RequestHandlerAvgIdlePercent)><>MeanRate
```
Comment (Akatroj) [inline resources/kafka-metrics.yaml:131]:
```suggestion
    # Sex) kafka.server<type=(KafkaRequestHandlerPool), name=(RequestHandlerAvgIdlePercent)><>MeanRate
```

**Review by Akatroj — CHANGES_REQUESTED** [review]

**Pull request merged**